### PR TITLE
nixVersions.nix_2_11: 2.11.0 -> 2.11.1

### PR DESCRIPTION
**Pull request by zowoq**
https://github.com/NixOS/nix/compare/2.11.0...2.11.1

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @mweinelt @vcunat

### CODE REVIEW

**Comment by Kranzes** [comment]
Should Nix release always be backported to stable?

**Comment by mweinelt** [comment]
I don't think we do, but it is likely fine for patch releases.

**Comment by Kranzes** [comment]
Should we not backport them in general?

**Comment by vcunat** [comment]
I would backport additions to `nixVersions` without bumping the default.  With patch releases (e.g. now) it seems appropriate to imply bumping the default, too.  Perhaps with some lag after getting to `nixos-unstable` channel, if there are no "urgent" fixes.

**Comment by vcunat** [comment]
What about adding a comment around the `stable = self.nix_2_11;` line about updating `nixos/modules/installer/tools/nix-fallback-paths.nix`?  It seems easy to forget and I don't even remember how to get the paths.

**Comment by SuperSandro2000** [comment]
We can safely backport this patch release because it does not touch any code that is used in nixos. There where just changes for the installer and fish support.

**Comment by SuperSandro2000** [comment]
> It seems easy to forget and I don't even remember how to get the paths.

I think thats because there does not seem to be any documentation in the repo. Maybe https://github.com/NixOS/nix/blob/master/maintainers/upload-release.pl but I am just guessing.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by vcunat** [comment]
Hmm yes, apparently it directly creates the file at
https://github.com/NixOS/nix/blob/master/maintainers/upload-release.pl#L217

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -106,8 +106,8 @@ in lib.makeExtensible (self: {
   };
 
   nix_2_11 = common {
-    version = "2.11.0";
-    sha256 = "sha256-9+rpYzI+SmxJn+EbYxjGv68Ucp22bdFUSy/4LkHkkDQ=";
+    version = "2.11.1";
+    sha256 = "sha256-qCV65kw09AG+EkdchDPq7RoeBznX0Q6Qa4yzPqobdOk=";
     patches = [
       ./patches/flaky-tests.patch
       (fetchpatch {
```
Comment (lovesegfault) [inline pkgs/tools/package-management/nix/default.nix:113]: Do we still need this patch or did it get included in 2.11.1?

Comment (vcunat) [reply pkgs/tools/package-management/nix/default.nix:113]: The patch still applies.

**Pull request merged**